cve_bin_tool/data_sources/nvd_source.py

```diff
@@ -247,27 +247,76 @@ def format_data_api2(self, all_cve_entries):
+                        "severity", "UNKNOWN"
+                    )
+                    if "cvssV2" in cve_item["impact"]["baseMetricV2"]:
+                        cve["score"] = cve_item["impact"]["baseMetricV2"]["cvssV2"].get(
+                            "baseScore", 0
+                        )
+                        cve["CVSS_vector"] = cve_item["impact"]["baseMetricV2"][
+                            "cvssV2"
+                        ].get("vectorString", "")
+
+            # Old data used "metrics" -- This section may need to be deleted
+            elif "metrics" in cve_item:
                 cve_cvss = cve_item["metrics"]
-            # Get CVSSv3 or CVSSv2 score
-            cvss_available = True
-            if "cvssMetricV31" in cve_cvss:
-                cvss_data = cve_cvss["cvssMetricV31"][0]["cvssData"]
-                cve["CVSS_version"] = 3
-            elif "cvssMetricV30" in cve_cvss:
-                cvss_data = cve_cvss["cvssMetricV30"][0]["cvssData"]
-                cve["CVSS_version"] = 3
-            elif "cvssMetricV2" in cve_cvss:
-                cvss_data = cve_cvss["cvssMetricV2"][0]["cvssData"]
-                cve["CVSS_version"] = 2
-            else:
-                LOGGER.debug(f"Unknown CVSS metrics field {cve_item['id']}")
-                cvss_available = False
-            if cvss_available:
-                cve["severity"] = cvss_data.get("baseSeverity", "UNKNOWN")
-                cve["score"] = cvss_data.get("baseScore", 0)
-                cve["CVSS_vector"] = cvss_data.get("vectorString", "")
+
+                # Get CVSSv3 or CVSSv2 score
+                cvss_available = True
+                if "cvssMetricV31" in cve_cvss:
+                    cvss_data = cve_cvss["cvssMetricV31"][0]["cvssData"]
+                    cve["CVSS_version"] = 3
+                elif "cvssMetricV30" in cve_cvss:
+                    cvss_data = cve_cvss["cvssMetricV30"][0]["cvssData"]
+                    cve["CVSS_version"] = 3
+                elif "cvssMetricV2" in cve_cvss:
+                    cvss_data = cve_cvss["cvssMetricV2"][0]["cvssData"]
+                    cve["CVSS_version"] = 2
+                else:
+                    cvss_available = False
+                if cvss_available:
+                    cve["severity"] = cvss_data.get("baseSeverity", "UNKNOWN")
+                    cve["score"] = cvss_data.get("baseScore", 0)
+                    cve["CVSS_vector"] = cvss_data.get("vectorString", "")
+            # End old metrics  section
+
+            # do some basic input validation checks
+            # severity should be alphanumeric
+            if not cve["severity"].isalnum():
+                self.logger.debug(
+                    f"Severity for {cve['id']} is invalid: {cve['severity']}"
+                )
+                cve["severity"] = re.sub(r"[\W]", "", cve["severity"])
+            # score should be numeric
+            if not cve["score"].isdecimal():
```

Comment:
The score is of the format X.Y e.g. 9.8. isdecimal() doesn't work with floats

Probably need  to use a regex like '[0-9]+\.[0-9]+'